dali/pipeline/operators/util/lookup_table.h

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATORS_UTIL_LOOKUP_TABLE_H_
+#define DALI_PIPELINE_OPERATORS_UTIL_LOOKUP_TABLE_H_
+
+#include <limits>
+#include <memory>
+#include <algorithm>
+#include <vector>
+#include "dali/pipeline/operators/operator.h"
+#include "dali/core/static_switch.h"
+#include "dali/core/convert.h"
+#include "dali/kernels/type_tag.h"
+
+namespace dali {
+
+template <typename Backend>
+class LookupTable : public Operator<Backend> {
+ public:
+  static constexpr size_t kLookupTableSize = 0x10000;
+
+  explicit inline LookupTable(const OpSpec &spec)
+    : Operator<Backend>(spec)
+    , output_type_(spec.GetArgument<DALIDataType>("output_dtype"))
+    , default_value_f_(spec.GetArgument<float>("default_value")) {
+    std::vector<int> keys;
+    int min_key = -1, max_key = -1;
+    std::vector<float> values_f;
+    if (spec.HasArgument("keys")) {
+      keys = spec.GetRepeatedArgument<int>("keys");
+      min_key = *std::min_element(keys.begin(), keys.end());
+      max_key = *std::max_element(keys.begin(), keys.end());
+      DALI_ENFORCE(min_key >= 0 && max_key <= static_cast<int>(kLookupTableSize),
+        "`keys` should be in the range [0, " + std::to_string(kLookupTableSize) + "]");
+    }
+    if (spec.HasArgument("values")) {
+      values_f = spec.GetRepeatedArgument<float>("values");
+    }
+    DALI_ENFORCE(keys.size() == values_f.size(),
+      "`keys` size should match `values` size");
+
+    TYPE_SWITCH(output_type_, dali::type2id, OutputType, (float, uint8_t, int16_t, int32_t), (
+        value_mem_.reset(new uint8_t[kLookupTableSize * sizeof(OutputType)]);
```

Comment:
Maybe type erasure with proper type allocation? I'm wary about upcasting to bigger sizes. 
```suggestion
        value_mem_.reset(new OutputType[kLookupTableSize], [](void* ptr)
        {
        	auto *to_delete = static_cast<OutputType*>(ptr);
            delete[] to_delete;
        }););
```

Reply:
will do